packages/vscode-extension/src/commonlib/codeFlowLogin.ts

```diff
@@ -356,3 +360,16 @@ export function ConvertTokenToJson(token: string): object {
   const buff = Buffer.from(array[1], "base64");
   return JSON.parse(buff.toString(UTF8));
 }
+
+export async function checkIsOnline(): Promise<boolean> {
+  try {
+    await new Promise<http.IncomingMessage>((resolve, reject) => {
+      const url = "https://login.microsoftonline.com/";
+      https.get(url, resolve).on("error", reject);
```

Comment:
Does `get` automatically follow redirect? A full `get` could be heavy, any `head` option?

Reply:
Changed to ```head``` request.